# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentPolicy.cpp
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 dom/base/nsContentPolicy.cpp
--- a/dom/base/nsContentPolicy.cpp	Tue May 21 22:51:54 2019 +0000
+++ b/dom/base/nsContentPolicy.cpp	Tue May 21 23:14:27 2019 +0000
@@ -70,7 +70,6 @@
                                              int16_t* decision) {
   nsContentPolicyType contentType = loadInfo->InternalContentPolicyType();
   nsCOMPtr<nsISupports> requestingContext = loadInfo->GetLoadingContext();
-  nsCOMPtr<nsIPrincipal> requestPrincipal = loadInfo->TriggeringPrincipal();
   nsCOMPtr<nsIURI> requestingLocation;
   nsCOMPtr<nsIPrincipal> loadingPrincipal = loadInfo->LoadingPrincipal();
   if (loadingPrincipal) {
@@ -98,18 +97,17 @@
    * iframes with an image as src. Get the uri from the dom node.
    * See bug 254510
    */
-  if (!requestingLocation) {
-    nsCOMPtr<mozilla::dom::Document> doc;
-    nsCOMPtr<nsIContent> node = do_QueryInterface(requestingContext);
-    if (node) {
-      doc = node->OwnerDoc();
-    }
-    if (!doc) {
-      doc = do_QueryInterface(requestingContext);
-    }
-    if (doc) {
-      requestingLocation = doc->GetDocumentURI();
-    }
+  nsCOMPtr<mozilla::dom::Document> doc;
+  nsCOMPtr<nsIContent> node = do_QueryInterface(requestingContext);
+  if (node) {
+    doc = node->OwnerDoc();
+  }
+  if (!doc) {
+    doc = do_QueryInterface(requestingContext);
+  }
+
+  if (!requestingLocation && doc) {
+    requestingLocation = doc->GetDocumentURI();
   }
 
   nsContentPolicyType externalType =
@@ -129,9 +127,8 @@
     window = do_QueryInterface(requestingContext);
   }
 
-  if (requestPrincipal) {
-    nsCOMPtr<nsIContentSecurityPolicy> csp;
-    requestPrincipal->GetCsp(getter_AddRefs(csp));
+  if (doc) {
+    nsCOMPtr<nsIContentSecurityPolicy> csp = doc->GetCsp();
     if (csp && window) {
       csp->EnsureEventTarget(
           window->EventTargetFor(mozilla::TaskCategory::Other));